### PR TITLE
Add lifecycle pv relcaim tc, improve lifecycle suite readability

### PR DESCRIPTION
**Pull request by shaior**
Automated 54201, 54202, 54203, 54204, 54206, 54207

Add lifecycle pv relcaim tc, improve lifecycle suite readability and consistency.

### CODE REVIEW

File: tests/utils/replicaset/replicaset.go [hunk 1]
```diff
@@ -38,3 +38,19 @@ func RedefineWithReplicaNumber(replicaSet *v1.ReplicaSet, replicasNumber int32)
 
 	return replicaSet
 }
+
+func RedefineWithVolume(replicaSet *v1.ReplicaSet, name string) *v1.ReplicaSet {
```
Comment (sebrandon1) [inline tests/utils/replicaset/replicaset.go:42]: I don't think you need to return the pointer here as it's already pass-by-reference.  Just modify the pointer in place.

File: tests/utils/pod/pod.go [hunk 2]
```diff
@@ -57,3 +57,19 @@ func RedefineWithLivenessProbe(pod *corev1.Pod) *corev1.Pod {
 
 	return pod
 }
+
+func RedefineWithVolume(pod *corev1.Pod, name string) *corev1.Pod {
```
Comment (sebrandon1) [inline tests/utils/pod/pod.go:61]: Same here, I don't think you need to return the pointer.

**Review by sebrandon1 — COMMENTED** [review]
Two comments.

File: tests/utils/pod/pod.go (hunk 2 above)
Comment (shaior) [reply tests/utils/pod/pod.go:61]: correct, thanks.
Will fix and upload a new version, I will also implement this across other functions.

Comment (sebrandon1) [reply tests/utils/pod/pod.go:61]: Agreed, it can be addressed later just pointing it out.

Comment (shaior) [reply tests/utils/pod/pod.go:61]: I agree and i think it's better to do it in a different PR, 
I will start working on one.

**Review by theemj — APPROVED** [review]

File: tests/lifecycle/tests/lifecycle_persistent_volume_reclaim_policy.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+		By("Define & create first pv")
+		persistentVolumea := persistentvolume.RedefineWithPVReclaimPolicy(
+			persistentvolume.DefinePersistentVolume(tsparams.TestPVName, tsparams.LifecycleNamespace), corev1.PersistentVolumeReclaimDelete)
+
+		err := tshelper.CreatePersistentVolume(persistentVolumea, tsparams.WaitingTime)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Define & create second pv")
+		persistentVolumeb := persistentvolume.RedefineWithPVReclaimPolicy(
+			persistentvolume.DefinePersistentVolume(tsparams.TestPVName, tsparams.LifecycleNamespace), corev1.PersistentVolumeReclaimRecycle)
```
Comment (greyerof) [inline tests/lifecycle/tests/lifecycle_persistent_volume_reclaim_policy.go:209]: Is this really working? This PV has the same name as the first one (TestPVName).

**Comment by shaior** [comment]
@greyerof as explained in the huddle, in the beforeEach block, I clean all resources that exist in the namespace, 
including PVs and PVCs. 
but, I was mistaken by relating the PVs deletion with the namespaces.clean, since PVs are cluster wide and not namespaced.
hence I moved the cleanPVs func to a different location.

As for the comment on the duplicated pv name, good catch, thank you, uploaded a new commit.

**Review by greyerof — APPROVED** [review]

**Pull request merged**